### DISCUSSION TITLE
age/length comps

### DISCUSSION THREAD

**Discussion by BenWilliams-NOAA** [afsc-assessments/afscdata#29] [Ideas]
@pete-hulson @zandjyo  Figured the comps will be a big enough issue that we should have a designated discussion space. 

To date:
 
 - comps are estimated using a number of different methods 
 - some can be pulled directly from GAP products, others need to bootstrap (and probably some other strategies as well).
 
We have a suite of one-off functions that people use for their specific assessment. 
In order to generalize this we should have a discussion on exactly what is needed/wanted so that we can build a function(s) that appropriately address the issues within the scope of afscdata/afscassess while avoiding and ad hoc/patchwork coding framework

**Comment by pete-hulson** [comment]
good idea - couple of thoughts here to kick this off:

1. we've got two primary categories of comps (fishery and survey) that we would need to pull data for, and
2. two primary categories of how these comps are expanded (either use pre-expanded data or do the expansion in a custom function)

1 above pertains to the afscdata package. for the survey comps my opinion would be to develop code to pull all the data/products that could possibly be needed (GAP expanded comps, cpue, strata, specimen, and length freq), out in the future this would allow for folks to use the expanded GAP comps, but also be able to integrate bootstrapping the iss (and, we have all this over in swo and surveyISS). for the fishery comps we need three essential data sets: the observer catch data, specimen, and length freq data - we're all doing these expansions on our own, but these would be the essential data to do that (albeit, there may be some other data out there that folks use, this would be a place to start i think).

2 above pertains to the afscassess package. we have the survey length & age comp expansion fcns (and the option to get the base expanded comps is in the surveyISS project), so i think this could be integrated without too much headache and have the base comps and iss be a standard output of this package (that one can choose to use if desired). the fishery comp expansions are not so straight forward. my initial thought was to get some code together that is nested within the fish_age_comp fcn in the input.R script and add a field that allows one to determine the expansion method - so, for example, i'd add an option to use the marginal ages, and option to use the marginal lengths and an alk, and then an option to expand the length comp by weighting with catch and then using an alk. i guess the approach here is that we contain things within what you've already developed, with the recognition that we may later decide to pull these out in stand alone scripts/fcns
